lib/saluki-env/src/autodiscovery/providers/local.rs

```diff
@@ -66,32 +76,76 @@ impl LocalAutoDiscoveryProvider {
     }
 }
 
+#[derive(Debug, Deserialize)]
+struct CheckConfig {
+    init_config: serde_yaml::Mapping,
+    instances: Vec<serde_yaml::Mapping>,
+}
+
 /// Parse a YAML file into a Config object
 async fn parse_config_file(path: &PathBuf) -> Result<(String, Config), GenericError> {
     let content = fs::read_to_string(path).await?;
 
+    let check_config: CheckConfig = match serde_yaml::from_str(&content) {
+        Ok(read) => read,
+        Err(e) => {
+            warn!("Can't decode yaml as check configuration: {}", content);
+            return Err(GenericError::from(e).context("Failed to decode yaml as check configuration."));
+        }
+    };
+
     // Build config ID from the file path
     let config_id = path
         .canonicalize()?
         .to_string_lossy()
         .to_string()
         .replace(['/', '\\'], "_");
 
+    let instances: Vec<HashMap<MetaString, MetaString>> = check_config
+        .instances
+        .into_iter()
+        .map(|instance| {
+            let mut result = HashMap::new();
+            for (key, value) in instance {
+                // Use serde_yaml to convert Value to String
+                if let Ok(value_str) = serde_yaml::to_string(&value) {
+                    // Strip leading and trailing quotes if present
+                    let clean_value = value_str.trim().trim_matches('"').to_string();
+                    let key_str = key.as_str().unwrap_or("unknown").to_string();
+                    result.insert(MetaString::from(key_str), MetaString::from(clean_value));
+                }
+            }
+            result
+        })
+        .collect();
+
+    let init_config = {
+        let mut result = HashMap::new();
+        for (key, value) in check_config.init_config {
+            if let Ok(value_str) = serde_yaml::to_string(&value) {
+                // Strip leading and trailing quotes if present
+                let clean_value = value_str.trim().trim_matches('"').to_string();
+                let key_str = key.as_str().unwrap_or("unknown").to_string();
+                result.insert(MetaString::from(key_str), MetaString::from(clean_value));
+            }
+        }
+        result
+    };
+
     // Create a Config
     let config = Config {
-        name: path.file_name().unwrap().to_string_lossy().to_string(),
-        event_type: EventType::Schedule,
-        init_config: content.as_bytes().to_vec(),
-        instances: Vec::new(), // Would parse from YAML in a real implementation
-        metric_config: Vec::new(),
-        logs_config: Vec::new(),
+        name: MetaString::from(path.file_name().unwrap().to_string_lossy().to_string()),
+        init_config,
+        instances,
+        metric_config: HashMap::new(),
+        logs_config: HashMap::new(),
         ad_identifiers: Vec::new(),
-        provider: "local".to_string(),
-        service_id: String::new(),
-        tagger_entity: String::new(),
+        provider: MetaString::from_static(""),
```

Comment:
Change these to `MetaString::empty()`.

Same thing.

Reply:
Done